### ISSUE TITLE
"unhandled opcode in isAArch64FrameOffsetLegal" for AArch64ISD::SVE_LD2_MERGE_ZERO

### ISSUE THREAD

**Issue by UsmanNadeem** [llvm/llvm-project#53781]
Crash seen in localstackalloc pass with llvm.aarch64.sve.ld2.nxv32i8

Error on SDNode: `nxv16i8,nxv16i8,ch = AArch64ISD::SVE_LD2_MERGE_ZERO t0, t33, FrameIndex:i64<0>`
MIR instruction: `%1:zpr2 = LD2B_IMM killed %0, %stack.0.mem_alloc, 0`

Reproducer: https://godbolt.org/z/aYPzccaWh
Started after: https://reviews.llvm.org/D112617 / 86972f111497bd15df8da181d0d7ac68b866320b

Dump:
```
unhandled opcode in isAArch64FrameOffsetLegal
UNREACHABLE executed at /root/llvm-project/llvm/lib/Target/AArch64/AArch64InstrInfo.cpp:4351!
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /opt/compiler-explorer/clang-assertions-trunk/bin/llc -o /app/output.s -x86-asm-syntax=intel -run-pass=localstackalloc -x=mir <source>
1.	Running pass 'Function Pass Manager' on module '<source>'.
2.	Running pass 'Local Stack Slot Allocation' on function '@test'
 #0 0x000055ef2f13f9df PrintStackTraceSignalHandler(void*) Signals.cpp:0:0
 #1 0x000055ef2f13d19d SignalHandler(int) Signals.cpp:0:0
 #2 0x00007fde5c3bf3c0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x153c0)
 #3 0x00007fde5be8e18b raise (/lib/x86_64-linux-gnu/libc.so.6+0x4618b)
 #4 0x00007fde5be6d859 abort (/lib/x86_64-linux-gnu/libc.so.6+0x25859)
 #5 0x000055ef2f082eaa (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x3082eaa)

 #6 0x000055ef2ce16d6b llvm::isAArch64FrameOffsetLegal(llvm::MachineInstr const&, llvm::StackOffset&, bool*, unsigned int*, long*) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0xe16d6b)
 #7 0x000055ef2ccd1b6b llvm::AArch64RegisterInfo::needsFrameBaseReg(llvm::MachineInstr*, long) const (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0xcd1b6b)
 #8 0x000055ef2e41ac9b (anonymous namespace)::LocalStackSlotPass::insertFrameReferenceRegisters(llvm::MachineFunction&) LocalStackSlotAllocation.cpp:0:0
 #9 0x000055ef2e41f558 (anonymous namespace)::LocalStackSlotPass::runOnMachineFunction(llvm::MachineFunction&) LocalStackSlotAllocation.cpp:0:0

#10 0x000055ef2e499286 llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x2499286)
#11 0x000055ef2e942c66 llvm::FPPassManager::runOnFunction(llvm::Function&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x2942c66)
#12 0x000055ef2e942fb9 llvm::FPPassManager::runOnModule(llvm::Module&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x2942fb9)
#13 0x000055ef2e9438f9 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0x29438f9)
#14 0x000055ef2cb57df8 compileModule(char**, llvm::LLVMContext&) llc.cpp:0:0
#15 0x000055ef2ca9618f main (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0xa9618f)
#16 0x00007fde5be6f0b3 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b3)
#17 0x000055ef2cb4e89a _start (/opt/compiler-explorer/clang-assertions-trunk/bin/llc+0xb4e89a)
Compiler returned: 139

**Comment by sdesmalen-arm** [comment]
This issue is currently being addressed by https://reviews.llvm.org/D119338

**Comment by kmclaughlin-arm** [comment]
This issue should be fixed after fc1b212
